app/src/main/java/me/tangobee/weathernaut/ui/SettingsFragment.kt

```diff
@@ -68,6 +72,28 @@ class SettingsFragment : Fragment() {
         binding.temperatureUnit.setOnClickListener {showTemperatureUnitPopup(binding.temperatureSpinner)}
         binding.windSpeedUnit.setOnClickListener {showWindSpeedUnitPopup(binding.windSpeedSpinner)}
         binding.atmosphericPressureUnit.setOnClickListener {showAtmosphericPressureUnitPopup(binding.atmosphericPressureSpinner)}
+        binding.language.setOnClickListener {showLanguagePopup(binding.languageSpinner)}
+        binding.language.apply {
+            val lang = context.resources.configuration.locale.language
+            when(lang){
+                "en" -> {
+                    binding.languageSpinner.text = "English"
+                    binding.flagIcon.setImageResource(R.drawable.country_flag_uk)
+                }
+                "es" -> {
+                    binding.languageSpinner.text = "Español"
+                    binding.flagIcon.setImageResource(R.drawable.country_flag_es)
+                }
+                "fr" -> {
+                    binding.languageSpinner.text = "Français"
+                    binding.flagIcon.setImageResource(R.drawable.country_flag_fr)
+                }
+                "de" -> {
+                    binding.languageSpinner.text = "Deutsch"
+                    binding.flagIcon.setImageResource(R.drawable.country_flag_de)
+                }
+            }
```

Comment:
Should we move this to a new method?

Reply:
this piece of code is only used once. So, making a function for only 1 usage would be unnecessary imo. What do you say?

Reply:
Yea, it will be used only once for now, but in the future, when we add more languages, the `when` statement will get longer and longer.